### PR TITLE
Add additional context with scram errors

### PR DESCRIPTION
**Pull request by chatton**
### All Submissions:

* [ ] Have you opened an Issue before filing this PR?
* [ ] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [ ] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?
* [ ] Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if such).

### CODE REVIEW

File: pkg/authentication/scram/scram.go [hunk 1]
```diff
@@ -271,7 +271,7 @@ func createScramCredentialsSecret(getUpdateCreator secret.GetUpdateCreator, mdbO
 func readExistingCredentials(secretGetter secret.Getter, mdbObjectKey types.NamespacedName, scramCredentialsSecretName string) (scramcredentials.ScramCreds, scramcredentials.ScramCreds, error) {
 	credentialsSecret, err := secretGetter.GetSecret(types.NamespacedName{Name: scramCredentialsSecretName, Namespace: mdbObjectKey.Namespace})
 	if err != nil {
-		return scramcredentials.ScramCreds{}, scramcredentials.ScramCreds{}, err
+		return scramcredentials.ScramCreds{}, scramcredentials.ScramCreds{}, errors.Errorf("could not get secret %s/%s: %s", scramCredentialsSecretName, mdbObjectKey.Namespace, err)
```
Comment (bznein) [inline pkg/authentication/scram/scram.go:274]: [q] shouldn't we invert here `scramCredentialsSecretName` and `mdbObjectKey.Namespace` for clarity?

File: pkg/authentication/scram/scram.go [hunk 2]
```diff
@@ -302,7 +302,7 @@ func convertMongoDBResourceUsersToAutomationConfigUsers(secretGetUpdateCreateDel
 	for _, u := range mdb.GetScramUsers() {
 		acUser, err := convertMongoDBUserToAutomationConfigUser(secretGetUpdateCreateDeleter, mdb.NamespacedName(), u)
 		if err != nil {
-			return nil, err
+			return nil, errors.Errorf("failed to convert user %s: %s", u.Username, err)
```
Comment (bznein) [inline pkg/authentication/scram/scram.go:305]: [nit] maybe add "to automation config user" to make it clearer?

**Review by bznein — APPROVED** [review]
LGTM minor comments only!

File: pkg/authentication/scram/scram.go (hunk 2 above)
Comment (chatton) [reply pkg/authentication/scram/scram.go:305]: yep good idea!

File: pkg/authentication/scram/scram.go (hunk 1 above)
Comment (chatton) [reply pkg/authentication/scram/scram.go:274]: ah yes! Thanks for spotting that

**Pull request merged**